### ISSUE TITLE
Keyboard input does not work for editing dates in Details component

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#713]
I think this didn't used to be the case, but I cannot change the dates using keyboard input. (This makes it completely inaccessible via screen reader, also.)

Also, if I use backspace to remove the date, keyboard input still cannot add a new date *and* the calendar widget breaks, putting it in an invalid state that cannot be fixed except by pressing 'Cancel'.
